core/flyout_base.js

```diff
@@ -55,6 +55,14 @@ Blockly.Flyout = function(workspaceOptions) {
   this.workspace_ = new Blockly.WorkspaceSvg(workspaceOptions);
   this.workspace_.isFlyout = true;
 
+  // When we create blocks for this workspace, instead of using the "optional" id
+  // make the default `id` the same as the `type` for easier re-use.
+  var newBlock = this.workspace_.newBlock;
+  this.workspace_.newBlock = function(type, id) {
+    // Use `type` if `id` isn't passed. `this` will be workspace.
+    return newBlock.call(this, proto, id || type);
```

Comment:
What is `proto` here?

Reply:
`type` but i changed that in a followup commit and forgot to change that, sorry. will fix